### PR TITLE
add v1.2 info

### PR DESCRIPTION
**Pull request by ritika-b**
- adds info about new turn-taking configuration
- adds callout for minimum spend
- adds auto-approval on mainnet callout
- add info on pre-execution validation

### CODE REVIEW

File: docs/chainlink-keepers/register-upkeep.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ After you register your contract as an Upkeep on the Keepers Registry, the Keepe
...
+     - The **Gas Limit** is the maximum amount of gas that will be used to execute your function on-chain. The Gas Limit of the [KeepersCounters.sol](/docs/chainlink-keepers/compatible-contracts#example-contract) example contract should be set to 200,000. We simulate `performUpkeep` during `checkUpkeep` and if the gas exceeds this limit the function will not execute on-chain. The gas limit cannot exceed the `callGasLimit` in the [configuration of the registry](/docs/chainlink-keepers/supported-networks/#configurations). *Note*: When your Upkeep is eligible for execution, the Chainlink Keepers Network will first simulate your `performUpkeep` and only execute an on-chain transaction if the `performUpkeep` does not revert. This will protect your Upkeep against possible logical errors in your contracts.
```
Comment (infiloop2) [inline docs/chainlink-keepers/register-upkeep.md:38]: There are actually 2 different gas limits, callGasLimit: limit for checkUpkeep and performGasLimit: limit for perform. 
The line "We simulate `performUpkeep` during `checkUpkeep` and if the gas exceeds this limit the function will not execute on-chain. The gas limit cannot exceed the `callGasLimit` in the [configuration of the registry](/docs/chainlink-keepers/supported-networks/#configurations)" is a bit confusing in this aspect, could we clarify this

Comment (infiloop2) [reply docs/chainlink-keepers/register-upkeep.md:38]: Also it is no longer exactly true that we simulate `performUpkeep` during `checkUpkeep`. (We do simulate performUpkeep before broadcasting a tx, but it's independent of checkUpkeep)

Comment (infiloop2) [reply docs/chainlink-keepers/register-upkeep.md:38]: We also mention it [here](https://documentation-git-fork-ritika-b-keepers-v2-chainlinklabs.vercel.app/docs/chainlink-keepers/compatible-contracts/#performupkeep-function)

**Comment by infiloop2** [comment]
Thanks @ritika-b for updating the docs. 

in 1.2 we also have the capability to update performGasLimit for an upkeep, maybe we can add that within `Manage your Upkeeps` section?

File: docs/chainlink-keepers/overview.md [hunk 2]
```diff
@@ -11,21 +11,21 @@ whatsnext:
...
+- **Upkeeps**: These are the jobs or tasks that developers want to execute on-chain, such as calling their smart contract function if a specific set of conditions have been met.
+- **Keepers registry**: The contract through which developers can [register](../register-upkeep/), and manage, their **Upkeeps**.
+- **Keepers**: Nodes in the Keepers Network that service registered and funded Upkeeps in the Keepers registry. Keepers use the same Node Operators as Chainlink Data Feeds, which secures billions of dollars in DeFi.
 
-The diagram below describes the architecture of the Keeper network. It is responsible for governing the actors on the network and compensating Keepers for performing successful Upkeeps. Clients can register for Upkeep and node operators can register as Keepers.
+The diagram below describes the architecture of the Keeper network. The Chainlink Keepers Registry is responsible for governing the actors on the network and compensating Keepers for performing successful Upkeeps. Developers can register their Upkeeps, and Node Operators can register as Keepers.
```
[the remark this answers is not in the document]
Comment (wentzeld) [reply docs/chainlink-keepers/overview.md]: No, this was there before though. I saw it and thought it must have been ok. Let's remove it.

[the remark this answers is not in the document]
Comment (ritika-b) [reply docs/chainlink-keepers/overview.md]: resolved

File: docs/chainlink-keepers/register-upkeep.md (hunk 1 above)
Comment (ritika-b) [reply docs/chainlink-keepers/register-upkeep.md]: resolved

**Comment by ritika-b** [comment]
Resolved performGasLimit comment

File: docs/chainlink-keepers/introduction.md [hunk 3]
```diff
@@ -25,34 +26,34 @@ To learn more about how the Chainlink Keepers Network automates your smart contr
...
     </div>
 
-1. [Register](../register-upkeep/) a new Upkeep in the [Chainlink Keepers App](https://keepers.chain.link) and select a **Time-based** trigger. Provide the address of your deployed contract, provide the ABI if it is not verified, and choose the function that you want to automate with relevant inputs.
+1. [Register](../job-scheduler/) a new Upkeep in the [Chainlink Keepers App](https://keepers.chain.link) and select a **Time-based** trigger. Provide the address of your deployed contract, provide the ABI if it is not verified, and choose the function that you want to automate alogn with the relevant function inputs, if any.
 
 1. Specify the time schedule using [CRON](../job-scheduler/#specifying-the-time-schedule).
 
-1. Complete the remaining details. Your email will be encrypted, but the project name will be visible to all. Your gas limit needs to include an extra [150K](../job-scheduler/#entering-upkeep-details).
+1. Complete the remaining details. Your Upkeep name will be visible to all, but your email and Project Name will not be visible to all. Your gas limit needs to include an extra [150K](../job-scheduler/#entering-upkeep-details) for execution.
```
Comment (dwightjl) [inline docs/chainlink-keepers/introduction.md:48]:
```suggestion
1. Complete the remaining details. Your upkeep name will be publicly visible, but your email and project name will not be publicly visible. Your gas limit needs to include an extra [150K](../job-scheduler/#entering-upkeep-details) for execution.
```

File: docs/chainlink-keepers/introduction.md [hunk 4]
```diff
@@ -62,25 +63,29 @@ To use a custom logic trigger, you will need to make your contract [Keepers-comp
         <a href="https://keepers.chain.link" >Open the Chainlink Keepers App</a>
     </div>
 
-1. [Register](../register-upkeep/) a new Upkeep in the [Chainlink Keepers App](https://keepers.chain.link) and select a **Custom logic** trigger. Provide the address of your contract and complete the remaining details. Your email will be encrypted, but the project name will be visible to all.
+1. [Register](../register-upkeep/) a new Upkeep in the [Chainlink Keepers App](https://keepers.chain.link) and select a **custom logic** trigger. Provide the address of your [Keepers-compatible](../compatible-contracts/) contract and complete the remaining details. Your Upkeep name will be visible to all, but your email and Project Name will not be visible to all. Ensure you specify the appropriate gas limit for your function to execute on chain.
```
Comment (dwightjl) [inline docs/chainlink-keepers/introduction.md:66]:
```suggestion
1. [Register](../register-upkeep/) a new Upkeep in the [Chainlink Keepers App](https://keepers.chain.link) and select a **custom logic** trigger. Provide the address of your [Keepers-compatible](../compatible-contracts/) contract and complete the remaining details. Your upkeep name will be publicly visible, but your email and project name will not be publicly visible. Ensure you specify the appropriate gas limit for your function to execute on chain.
```

File: docs/chainlink-keepers/introduction.md [hunk 5]
```diff
@@ -62,25 +63,29 @@ To use a custom logic trigger, you will need to make your contract [Keepers-comp
...
 > ❗️ **WARNING**
 >
 > Do **NOT** attempt to send LINK to your contract like you do with [VRF](../../get-a-random-number/). For Chainlink Keepers, contracts are funded via the registry rather than within your contract.
 
-> ⚠️ Registration Onboarding Note
->
-> Registrations on a testnet will be approved immediately. Mainnet registrations will be reviewed by our onboarding team before being approved. We are working towards a fully self-serve model.
+  > 🚧 Funding Upkeep
+    >
+    > You should fund your contract with more LINK than you anticipate you will need. The network will not check or perform your upkeep if your balance is too low based on current exchange rates. View the [Keepers Economics](../keeper-economics) page to learn more about the cost of using Keepers.
```
Comment (dwightjl) [inline docs/chainlink-keepers/introduction.md:80]:
```suggestion
    > Fund your contract with more LINK than you anticipate you will need. The network will not check or perform your upkeep if your balance is too low based on current exchange rates. View the [Keepers Economics](../keeper-economics) page to learn more about the cost of using Keepers.
```

File: docs/chainlink-keepers/introduction.md [hunk 6]
```diff
@@ -62,25 +63,29 @@ To use a custom logic trigger, you will need to make your contract [Keepers-comp
...
+
+  > 🚧 ERC677 Link
+    >
+    > For registration on Mainnet, you must use ERC-677 LINK. Many token bridges give you ERC-20 LINK tokens. Use PegSwap to [convert Chainlink tokens (LINK) to be ERC-677 compatible](https://pegswap.chain.link/). To register on a supported testnet, get [LINK](../../link-token-contracts/) from [faucets.chain.link](https://faucets.chain.link/).
```
Comment (dwightjl) [inline docs/chainlink-keepers/introduction.md:84]:
```suggestion
    > For registration on a mainnet, you must use ERC-677 LINK. Many token bridges give you ERC-20 LINK tokens. Use PegSwap to [convert Chainlink tokens (LINK) to be ERC-677 compatible](https://pegswap.chain.link/). To register on a supported testnet, get [LINK](../../link-token-contracts/) from [faucets.chain.link](https://faucets.chain.link/).
```

**Review by dwightjl — CHANGES_REQUESTED** [review]

**Comment by dwightjl** [comment]
Rebased to main

**Comment by dwightjl** [comment]
Closing in favor of https://github.com/smartcontractkit/documentation/pull/888

**Pull request closed without merging**